src/FasterReflection/ReflectionMetadataBuilder.cs

```diff
@@ -131,12 +132,34 @@ public ReflectionMetadataResult Build()
             }
             if (exceptions.Count > 0)
             {
-                DisposeAssemblies(assemblies);
-                throw new AggregateException("One or more assemblies failed to load", exceptions);
+                try
```

Comment:
Instead of doing this after assembly processing, move the `if (ignoreDuplicateAssemblies && ex.HResult != -2146233079)` to the `exceptions.Add` line above, and leave the dispose & throw code as-is. Also please add a meaningful constant for the `HResult`.